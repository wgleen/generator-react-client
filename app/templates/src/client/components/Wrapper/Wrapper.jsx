import React from 'react'
import { Provider } from 'react-redux'
import {
  store
} from '../../lib/store'

const Wrapper = props => (
  <Provider store={store}>
    {props.children}
  </Provider>
)

export default Wrapper