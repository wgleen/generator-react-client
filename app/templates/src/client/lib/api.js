import axios from 'axios'
import config from '../config'
import { todosMock } from '../mocks/todosMock.js'

const api = axios.create({
  baseURL: config.api.baseURL,
  timeout: 10000
})

const generateShowResource = (resource, mock) =>
  process.env.ENV == 'test' ?
    () => ({
      then: (callback => {
        callback({
          data: mock
        })

        return ({ catch: () => null })
      })
    }) :
    resource()

const generateCreateResource = (resource, mock, data) => {
  if (process.env.ENV == 'test') {
    mock.push({
      id: mock.length + 1,
      ...data
    })

    return () => ({
      then: (callback => {
        callback({ data })

        return ({ catch: () => null })
      })
    })
  } else return resource()
}

export const createTodosResource = data => 
  generateCreateResource(
    () => api.post('todos', data),
    todosMock, data
  )(data)

export const getTodosResource = generateShowResource(() =>
  () => api.get('todos'),
  todosMock
)

export default api