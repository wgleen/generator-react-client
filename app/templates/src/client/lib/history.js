import { createBrowserHistory } from 'history'

const _history = createBrowserHistory()

_history.listen(() => {
  window.scrollTo(0, 0)
})

export default _history