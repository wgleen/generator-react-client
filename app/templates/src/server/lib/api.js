import axios from 'axios'
import config from '../config'

const api = axios.create({
  baseURL: config.api.baseURL,
  timeout: 10000
})

export default api
