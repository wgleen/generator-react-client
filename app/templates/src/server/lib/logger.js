import winston, { createLogger } from 'winston'
import morgan from 'morgan'

const level = process.env.LOG_LEVEL || 'debug'

const logger = createLogger({
  transports: [
    new winston.transports.Console({
      level,
      timestamp: () => (new Date()).toISOString()
    })
  ],
  exitOnError: false
})

logger.stream = { write: message => logger.info(message) }

export default morgan('combined', { stream: logger.stream })
