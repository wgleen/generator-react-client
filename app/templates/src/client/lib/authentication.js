import api from './api'

export const setAccessToken = (token) => {
  if (token) {
    api.defaults.headers.common.Authorization = token
    localStorage.setItem('accessToken', token)
  } else {
    delete api.defaults.headers.common.Authorization
    delete localStorage.accessToken
    delete localStorage.keepLogged
  }
}

export const keepLoggedVerify = () => {
  window.onbeforeunload = () => {
    if (!localStorage.getItem('keepLogged')) return setAccessToken()

    return false
  }
}

export const keepLogged = value => localStorage.setItem('keepLogged', value)
