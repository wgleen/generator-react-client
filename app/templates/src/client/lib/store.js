import {
  createStore,
  applyMiddleware,
  compose
} from 'redux'
import thunk from 'redux-thunk'
import multi from 'redux-multi'
import promise from 'redux-promise'
import { composeWithDevTools } from 'redux-devtools-extension'
import { connectRouter, routerMiddleware } from 'connected-react-router'
import rootReducer from '../reducers'
import history from './history'

const middlewares = applyMiddleware(
  routerMiddleware(history),
  thunk,
  multi,
  promise
)

const store = createStore(
  connectRouter(history)(rootReducer),
  compose(process.env.ENV === 'development' ? composeWithDevTools(middlewares) : middlewares)
)

export default store
