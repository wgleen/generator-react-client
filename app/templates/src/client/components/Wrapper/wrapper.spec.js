import React from 'react'
import { mount } from 'enzyme'
import Wrapper from '../Wrapper/Wrapper'
import { store } from '../../lib/store'

describe('<Wrapper />', () => {
  it('should render correctly', () => {
    let component = mount(
      <Wrapper><div /></Wrapper>
    )

    expect(component).toMatchSnapshot()
    
    component.unmount()
  })

  it('should contain Provider component with store prop', () => {
    let component = mount(
      <Wrapper><div /></Wrapper>
    )

    expect(
      component
        .find('Provider')
          .props()
          .store
    ).toEqual(store)
    
    component.unmount()
  })
})