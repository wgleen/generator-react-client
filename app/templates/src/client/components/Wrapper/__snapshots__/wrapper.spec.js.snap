// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wrapper /> should render correctly 1`] = `
<Wrapper>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <div />
  </Provider>
</Wrapper>
`;
